packages/core/src/electron-main/electron-main-application.ts

```diff
@@ -626,21 +637,29 @@ export class ElectronMainApplication {
 
     protected restart(id: number): void {
         this.restarting = true;
-        const window = BrowserWindow.fromId(id);
-        window.on('closed', async () => {
+        const browserWindow = BrowserWindow.fromId(id);
+        // eslint-disable-next-line no-null/no-null
+        if (browserWindow === null) {
+            throw new Error(`no BrowserWindow with id: ${id}`);
+        }
+        browserWindow.on('closed', async () => {
             await this.launch({
                 secondInstance: false,
                 argv: this.processArgv.getProcessArgvWithoutBin(process.argv),
                 cwd: process.cwd()
             });
             this.restarting = false;
         });
-        this.handleStopRequest(window, () => this.doCloseWindow(window), StopReason.Restart);
+        this.handleStopRequest(browserWindow, () => this.doCloseWindow(browserWindow), StopReason.Restart);
     }
 
     protected async handleReload(event: Electron.IpcMainEvent): Promise<void> {
-        const window = BrowserWindow.fromId(event.sender.id);
-        this.reload(window);
+        const browserWindow = BrowserWindow.fromId(event.sender.id);
+        // eslint-disable-next-line no-null/no-null
+        if (browserWindow === null) {
+            throw new Error(`no BrowserWindow with id: ${event.sender.id}`);
+        }
```

Comment:
Minor: We should be able to not use `null`
```suggestion
        if (!browserWindow) {
            throw new Error(`no BrowserWindow with id: ${id}`);
        }
```